src/excmds_background.ts

```diff
@@ -179,6 +136,40 @@ export function open(url: string) {
     url = (hasScheme(url)? "" : "http://") + url
     messageActiveTab("open", [url])
 }
+export async function winopen(...args){
+    let address: string
+    const createData = {}
+    if (args[0] === "-private") {
+        createData['incognito'] = true
+        if (args[1]) address = args[1]
+    }
+    else if (args[0]) address = args[0]
+    if (address) createData['url'] = (hasScheme(address)? "" : "http://") + address
+    browser.windows.create(createData)
+}
+export async function winclose() {
+    browser.windows.remove((await browser.windows.getCurrent()).id)
+}
+export async function tabopen(address?: string){
+    if (address) address = (hasScheme(address)? "" : "http://") + address
```

Comment:
There are few more frequently used schemes supported by browsers. Namely `https://` and `file://`. `https://` is usually solved by auto redirects or plugins. But `file://` is the pain in the ass. I would therefore propose detecting, whether the `address` variable contains any valid HTTP address (i.e. does not begin with `/` nor `./` nor `../`). If it's a valid HTTP address, then proceed like now, but if it's not, then I would add the `file://` scheme instead.